noir-projects/noir-protocol-circuits/crates/types/src/traits.nr

```diff
@@ -184,3 +184,201 @@ impl<let N: u32> Deserialize<N> for [Field; N] {
         fields
     }
 }
+
+/// A trait for types that can be deserialized from a calldata field array
+#[derive_via(derive_from_calldata)]
+pub trait FromCalldata<let N: u32> {
```

Comment:
I wonder if you can't use a Reader instead of having to define all these: https://github.com/AztecProtocol/aztec-packages/blob/master/noir-projects/noir-protocol-circuits/crates/types/src/utils/reader.nr

Reply:
BTW when compiling all contracts I'm getting tons of "missing FromCalldata implementation" errors. IDK if it's feasible to reimplement all those.

```
error: No matching impl found for `EmbeddedCurvePoint: FromCalldata<_>`
   ┌─ contracts/token_contract/src/main.nr:14:3
   │
14 │ #[aztec]
   │   ----- No impl for `EmbeddedCurvePoint: FromCalldata<_>`
   │

error: No matching impl found for `str<31>: FromCalldata<_>`
   ┌─ contracts/token_contract/src/main.nr:14:3
   │
14 │ #[aztec]
   │   ----- No impl for `str<31>: FromCalldata<_>`
   │

error: Type annotation needed
  ┌─ contracts/auth_contract/src/main.nr:7:3
  │
7 │ #[aztec]
  │   ----- Could not determine type of generic argument
  │
  = Call stack:
    1. contracts/auth_contract/src/main.nr:7:3
 
...
```

Reply:
Oh, I was just trying things on the avm test contract. I'll check all contracts and add the missing trait impls.